crates/shared/src/event_handling.rs

```diff
@@ -202,6 +233,35 @@ where
     }
 }
 
+fn detect_reorg_path(
+    handled_blocks: &Vec<BlockNumberHash>,
+    current_blocks: &Vec<BlockNumberHash>,
+) -> Result<RangeInclusive<BlockNumberHash>> {
+    ensure!(!handled_blocks.is_empty() && !current_blocks.is_empty());
+
+    // in most cases, current_blocks = handled_blocks + 1 newest block
+    // therefore, is it more efficient to put the handled_blocks in outer loop,
+    // so everything finishes in only two iterations.
+    for handled_block in handled_blocks.iter().rev() {
+        for current_block in current_blocks.iter().rev() {
+            if current_block.0 == handled_block.0 && current_block.1 == handled_block.1 {
+                // found the same block in both lists, now we know the common ancestor
+                return Ok(*current_block..=*current_blocks.last().unwrap());
```

Comment:
Yes, I put a common ancestor in the list even if it's not mandatory. Seems like a good tradeoff for code simplicity. I might reconsider though...